winsup/cygwin/exceptions.cc

```diff
@@ -1550,8 +1551,23 @@ sigpacket::process ()
 dosig:
   if (have_execed)
     {
-      sigproc_printf ("terminating captive process");
-      TerminateProcess (ch_spawn, sigExeced = si.si_signo);
+      switch (si.si_signo)
+        {
+        case SIGUSR1:
+        case SIGUSR2:
+        case SIGCONT:
+        case SIGSTOP:
+        case SIGTSTP:
+        case SIGTTIN:
+        case SIGTTOU:
+          system_printf ("Suppressing signal %d to win32 process (pid %u)",
+              (int)si.si_signo, (unsigned int)GetProcessId(ch_spawn));
+          goto done;
+        default:
+          sigproc_printf ("terminating captive process");
+          rc = exit_process_tree (ch_spawn, 128 + (sigExeced = si.si_signo));
```

Comment:
@dscho I am confused here. Why should `128` be added here while the previous `TerminateProcess` didn't have them? Can you explain?

Reply:
I am asking this because doing 
```cmd
test CtrlRoutine 0 9032
```
seem's to kill the process with traceback

while 
```cmd
test CtrlRoutine 1 9032
```
force kill it.

But
```cmd
test CtrlRoutine 128 9032
```
doesn't do anything.

Maybe that's the problem things not working?

Reply:
Maybe I am just confused.

Reply:
The convention that signals result in exit codes equaling to 128 plus the signal number is explained e.g. here: https://stackoverflow.com/questions/1101957/are-there-any-standard-exit-status-codes-in-linux/1535733#1535733